pbincli/actions.py

```diff
@@ -1,217 +1,150 @@
 
         result = api_client.get(pasteid)
     else:
         print("PBinCLI error: Incorrect request")
-        sys.exit(1)
+        exit(1)
 
     if args.debug: print("Response:\t{}\n".format(result))
 
-    try:
-        result = json.loads(result)
-    except ValueError as e:
-        print("PBinCLI Error: {}".format(e))
-        sys.exit(1)
-
     if 'status' in result and not result['status']:
-        print("Paste received! Text inside:")
-        data = json.loads(result['data'])
-
-        if args.debug: print("Text:\t{}\n".format(data))
-
-        text = SJCL().decrypt(data, password)
+        print("Paste received!")
+        if args.debug: print("Message:\t{}\nAuthentication data:\t{}".format(result['ct'], result['adata']))
 
+        paste = Paste.getInstance(version = result['v'] if 'v' in result else 1)
+        paste.setPaste(result)
+        paste.setHash(passphrase)
+        if args.password:
+            paste.setPassword(args.password)
+        paste.decrypt()
+        text = paste.getText()
         if args.debug: print("Decoded text size: {}\n".format(len(text)))
 
-        if len(text):
-            print("{}\n".format(decompress(text.decode())))
+        check_writable('paste.txt')
+        with open('paste.txt', 'wb') as f:
+            f.write(text)
+            f.close
 
-            check_writable("paste.txt")
-            with open("paste.txt", "wb") as f:
-                f.write(decompress(text.decode()))
-                f.close
+        attachment, attachment_name = paste.getAttachment()
+        if attachment:
+            print('Found file, attached to paste. Decoding it and saving')
 
-        if 'attachment' in result and 'attachmentname' in result:
-            print("Found file, attached to paste. Decoding it and saving")
+            if args.debug: print("Name:\t{}\nData:\t{}".format(attachment_name, attachment))
 
-            cipherfile = json.loads(result['attachment']) 
-            cipherfilename = json.loads(result['attachmentname'])
-
-            if args.debug: print("Name:\t{}\nData:\t{}".format(cipherfilename, cipherfile))
-
-            attachmentf = SJCL().decrypt(cipherfile, password)
-            attachmentname = SJCL().decrypt(cipherfilename, password)
-
-            attachment = decompress(attachmentf.decode('utf-8')).decode('utf-8').split(',', 1)[1]
-            file = b64decode(attachment)
-            filename = decompress(attachmentname.decode('utf-8')).decode('utf-8')
-
-            print("Filename:\t{}\n".format(filename))
-
-            check_writable(filename)
-            with open(filename, "wb") as f:
-                f.write(file)
+            check_writable(attachment_name)
+            with open(attachment_name, 'wb') as f:
+                f.write(attachment)
                 f.close
 
-        if 'burnafterreading' in result['meta'] and result['meta']['burnafterreading']:
+        # burn after reading via API, only required for PrivateBin < 1.2
```

Comment:
How we can detect if PBin version is < 1.2?

Reply:
That is correct. Hence I propose to send that burnafterreading delete for all v1 pastes just in case it happens to be a 1.2.1 or 1.2.0 instance.

Note for future reference: The version string in the HTML isn't a reliable indicator, as some folks change it in their server and not all templates have it in the same HTML element.

Reply:
I'm asking that because I found that 1.2.1 returns empty response, which causes exception.

And I made exception handler to prevent this:
```diff
-    def delete(self, pasteid, token):
-        request = {'pasteid':pasteid,'deletetoken':token}
-        r = requests.post(url = self.server, headers = self.headers, proxies = self.proxy, data = request)
-        return r.text
+    def delete(self, request):
+        result = requests.post(
+            url = self.server,
+            headers = self.headers,
+            proxies = self.proxy,
+            data = request)
+
+        # using try as workaround for versions < 1.3 due to we cant detect
+        # if server used version 1.2, where auto-deletion is added
+        try:
+            return result.json()
+        except ValueError as e:
+            # unable parse response as json because it can be empty (1.2), so simulate correct answer
+            from json import loads as json_loads
+            return json_loads('{"status":0}')
```